src/Poisson.jl

```diff
@@ -163,7 +163,7 @@ function solver!(p::Poisson;log=false,tol=1e-4,itmx=1e3)
     log && (res = [r₂])
     nᵖ=0
     while r₂>tol && nᵖ<itmx
-        smooth!(p); r₂ = L₂(p)
+        pcg!(p); r₂ = L₂(p)
```

Comment:
why this change?

Reply:
Because when PCG is used directly as a solver (not as a smoother) it can get stuck without making any progress if we don't allow a larger error margin. The GMG should just quit and try something else.